### PR TITLE
Trying to OO this code and make it more sane and readable.

### PR DESCRIPTION
**Pull request by mikedarzen**

### CODE REVIEW

**Review by NicJudsonZenPlanner — APPROVED** [review]
wow, been years since I looked at C++ code.  Nicely organized 👍

File: ESP8266 Beacon Client/BeaconFader.cpp [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+}
+
+void BeaconFader::init(BeaconColor sourceColor, BeaconColor destinationColor) {
+  this -> sourceColor = sourceColor;
+  this -> destinationColor = destinationColor;
+  red = sourceColor.getRed();
+  green = sourceColor.getGreen();
+  blue = sourceColor.getBlue();
+  white = sourceColor.getWhite();
+  rRatio = ((double)(sourceColor.getRed() - destinationColor.getRed())/((double)255)) * -1;
```
Comment (petrig) [inline ESP8266 Beacon Client/BeaconFader.cpp:19]: Could just set the opposite.
```suggestion
  rRatio = -((double)(sourceColor.getRed() - destinationColor.getRed())/((double)255));
```

File: ESP8266 Beacon Client/BeaconFader.cpp [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  bRatio = ((double)(sourceColor.getBlue() - destinationColor.getBlue())/((double)255)) * -1;
+  wRatio = ((double)(sourceColor.getWhite() - destinationColor.getWhite())/((double)255)) * -1;
+}
+
+bool BeaconFader::isComplete() {
+  return getCurrentColor().equals(destinationColor);
+}
+
+BeaconColor BeaconFader::getNext() {
+  red = red + rRatio;
```
Comment (petrig) [inline ESP8266 Beacon Client/BeaconFader.cpp:30]: Could use combination operator.
```suggestion
  red += rRatio;
```

File: ESP8266 Beacon Client/BeaconFader.cpp [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+  green = green + gRatio;
+  blue = blue + bRatio;
+  white = white + wRatio;
+  normalize();
+  return getCurrentColor();
+}
+
+void BeaconFader::normalize() {
+  if (red > 255) red = 255;
+  if (red < 0) red = 0;
```
Comment (petrig) [inline ESP8266 Beacon Client/BeaconFader.cpp:40]: No braces and unnecessary tests.
```suggestion
  if (red > 255) { red = 255 } else if (red < 0) { red = 0 };
```

File: ESP8266 Beacon Client/HandleSocketIo.ino [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+                  (int) (currentColor & 0x000000FF)        //white
+                );
+  }
+  ledController.setProgram(frequency, colorCount, colors, lightPattern, colorPattern);
+}
+
+LightPattern translateLightPattern(String lightPattern) {
+  if (lightPattern == "fade")
+    return LightPattern::FADE;
+  else if (lightPattern == "strobe") {
```
Comment (petrig) [inline ESP8266 Beacon Client/HandleSocketIo.ino:75]:
```suggestion
  if (lightPattern == "fade") {
    return LightPattern::FADE;
  } else if (lightPattern == "strobe") {
```

**Review by petrig — APPROVED** [review]
Recommend always using braces with if statements.

**Pull request merged**